### PR TITLE
introduced benchmark script for embeddings endpoint

### PR DESCRIPTION
**Pull request by dtrawins**
### 🛠 Summary

CVS-156492
Introduced benchmark script for embeddings endpoint

### 🧪 Checklist

- [ ] Unit tests added.
- [x] The documentation updated.
- [ ] Change follows security best practices.
``

### CODE REVIEW

File: demos/embeddings/README.md [hunk 1]
```diff
@@ -155,7 +155,39 @@ It will report results like `Similarity score as cos_sim 0.97654650115054`.
 
 ## Benchmarking feature extraction
 
-TBD
+An asynchronous benchmarking client can be use to access the model server performance with various load conditions:
+```bash
```
Comment (porlows1) [inline demos/embeddings/README.md:159]: From automatic tests point of view at this moment current directory is model_server.
I suggest to add **cd demos/embeddings** command here

**Review by porlows1 — CHANGES_REQUESTED** [review]

**Review by porlows1 — APPROVED** [review]
Approved but please consider adding required cd command to embeddngs/README.md

File: demos/benchmark/embeddings/benchmark_embeddings.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+from dataclasses import dataclass, field
+from typing import AsyncGenerator, List, Optional, Union
+from transformers import AutoTokenizer
+import argparse
+import aiohttp
+
+AIOHTTP_TIMEOUT = aiohttp.ClientTimeout(total=6 * 60 * 60)
+
+parser = argparse.ArgumentParser(description='Run benchmark for embeddings endpoints')
+parser.add_argument('--dataset', required=False, default='Cohere/wikipedia-22-12-simple-embeddings', help='Dataset for load generation from HF or a keyword synthetic_short or synthetic_long', dest='dataset')
```
Comment (dkalinowski) [inline demos/benchmark/embeddings/benchmark_embeddings.py:36]: validate possible list of datasets? display possible options, from line 42-53 i can see that there are custom ones (not from HF) - synthetic_short and synthetic_long which user may not be able to see by running --help

File: demos/benchmark/embeddings/benchmark_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+parser.add_argument('--api_url', required=False, default='http://localhost:8000/v3/embeddings', help='API URL for embeddings endpoint', dest='api_url')
+parser.add_argument('--model', required=False, default='Alibaba-NLP/gte-large-en-v1.5', help='HF model name', dest='model')
+parser.add_argument('--request_rate', required=False, default='inf', help='Request rate', dest='request_rate')
+parser.add_argument('--batch_size', required=False, type=int, default=16, help='Batch size', dest='batch_size')
+
+args = vars(parser.parse_args())
+docs = Dataset.from_dict({})
+if args["dataset"] == 'synthetic_short':
+    dummy_text = "hi " * 5
```
Comment (dkalinowski) [inline demos/benchmark/embeddings/benchmark_embeddings.py:45]: 5 and 500 could be a parameter to the script, `--synthetic_length` ?

File: demos/benchmark/embeddings/benchmark_embeddings.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    for i in range(1000):
+        docs = docs.add_item({"text":dummy_text})
+elif args["dataset"] == 'synthetic_long':
+    dummy_text = "hi " * 500
+    for i in range(1000):
+        docs = docs.add_item({"text":dummy_text})
+else:
+    docs = load_dataset(args["dataset"],split='train[:1000]')
```
Comment (dkalinowski) [inline demos/benchmark/embeddings/benchmark_embeddings.py:53]: Add log for each if, which dataset was picked

File: demos/benchmark/embeddings/benchmark_embeddings.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+print("Number of documents:",len(docs))
+
+batch_size = args['batch_size']
+
+def get_tokens(docs, model):
```
Comment (dkalinowski) [inline demos/benchmark/embeddings/benchmark_embeddings.py:59]:
```suggestion
def get_number_of_tokens(docs, model):
```

File: demos/benchmark/embeddings/benchmark_embeddings.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    documents = documents_all.iter(batch_size=batch_size)
+    for request in documents:
+        yield request["text"]
+        if request_rate == float("inf"):
+            # If the request rate is infinity, then we don't need to wait.
+            continue
+        # Sample the request interval from the exponential distribution.
+        interval = np.random.exponential(1.0 / request_rate)
+        # The next request will be sent after the interval.
+        await asyncio.sleep(interval)
```
Comment (dkalinowski) [inline demos/benchmark/embeddings/benchmark_embeddings.py:139]: Can you add reference what benchmarking code is it based of off? I think it is vLLM's benchmark_serving,py but I cant remember

**Pull request merged**